sonic_installer/main.py

```diff
@@ -106,9 +111,15 @@ def set_default_image(image):
     if get_running_image_type() == IMAGE_TYPE_ABOOT:
         image_path = aboot_image_path(image)
         aboot_boot_config_set(SWI=image_path, SWI_DEFAULT=image_path)
-    else:
+    elif BOOTLOADER == BOOTLOADER_TYPE_GRUB:
         command = 'grub-set-default --boot-directory=' + HOST_PATH + ' ' + str(images.index(image))
         run_command(command)
+    elif BOOTLOADER == BOOTLOADER_TYPE_UBOOT:
+        if image in images[0]:
+            run_command('/usr/bin/fw_setenv boot_next "run sonic_image_1"')
```

Comment:
Using fw_printenv/fw_setenv to get/set a uboot environment variable is platform specific.
I think platform specific code shouldn't appear here. Can you move it to platform directory.